tests/ci.js

```diff
@@ -10,70 +10,60 @@ const elmTestVersion = packageInfo.version;
 
 function run(testFile, clearCache) {
   if (clearCache !== false) {
-    console.log(
-      '\nClearing ' + path.join(process.cwd(), 'elm-stuff') + ' prior to run'
+    shell.echo(
+      'Clearing ' + path.join(process.cwd(), 'elm-stuff') + ' prior to run'
     );
     shell.rm('-rf', 'elm-stuff');
   }
 
-  if (!testFile) {
-    let cmd = [elmTest, '--color'].join(' ');
+  let cmd = [elmTest, ...(testFile ? [testFile] : []), '--color'].join(' ');
 
-    shell.echo('Running: ' + cmd);
-    return shell.exec(cmd, spawnOpts).code;
-  } else {
-    let cmd = [elmTest, testFile, '--color'].join(' ');
-
-    shell.echo('Running: ' + cmd);
-    return shell.exec(cmd, spawnOpts).code;
-  }
+  shell.echo('Running: ' + cmd);
+  return shell.exec(cmd, spawnOpts);
 }
 
-function assertTestErrored(testfile, clearCache) {
-  var code = run(testfile, clearCache);
-  if (code !== 1) {
-    shell.exec(
-      'echo ' +
-        filename +
-        ': error: ' +
-        (testfile ? testfile + ': ' : '') +
-        'expected tests to exit with ERROR exit code, not exit code' +
-        code +
-        ' >&2'
-    );
-    shell.exit(1);
+function assertTestErrored(testFile, clearCache) {
+  const result = run(testFile, clearCache);
+  if (result.code !== 1) {
+    fail('Expected tests to exit with exit code 1.', result);
   }
 }
 
-function assertTestFailure(testfile, clearCache) {
-  var code = run(testfile, clearCache);
-
-  if (code < 2) {
-    shell.exec(
-      'echo ' +
-        filename +
-        ': error: ' +
-        (testfile ? testfile + ': ' : '') +
-        'expected tests to fail >&2'
-    );
-    shell.exit(1);
+function assertTestFailure(testFile, clearCache) {
+  const result = run(testFile, clearCache);
+  if (result.code < 2) {
+    fail('Expected tests to fail with exit code 2 or higher.', result);
   }
 }
 
 function assertTestSuccess(testFile, clearCache) {
-  var code = run(testFile, clearCache);
-  if (code !== 0) {
-    shell.exec(
-      'echo ' +
-        filename +
-        ': ERROR: ' +
-        (testFile ? testFile + ': ' : '') +
-        'Expected tests to pass >&2'
-    );
-    shell.exit(1);
+  const result = run(testFile, clearCache);
+  if (result.code !== 0) {
+    fail('Expected tests to pass with exit code 0.', result);
   }
 }
 
+function fail(message, result) {
+  shell.echo();
+  shell.echo(`######### ERROR`);
+  shell.echo(
+    "The last elm-test run above didn't run as expected. Details below."
+  );
+  shell.echo();
+  shell.echo('### stdout');
+  shell.echo(result.stdout || '(no stdout)');
+  shell.echo();
+  shell.echo('### stderr');
+  shell.echo(result.stderr || '(no stderr)');
+  shell.echo();
+  shell.echo('### message');
+  shell.echo(message);
+  shell.echo(`Exit code: ${result.code}. See stdout and stderr above.`);
+  shell.echo(`This message comes from tests/${filename}`);
+  shell.echo();
+  shell.exit(1);
+}
```

Comment:
This helped a lot when trying to get all tests to pass again.

Reply:
This will make my life my easier too going forward!

Reply:
Merged in https://github.com/rtfeldman/node-test-runner/pull/448